src/librustc_session/options.rs

```diff
@@ -948,4 +948,6 @@ options! {DebuggingOptions, DebuggingSetter, basic_debugging_options,
          (such as entering an empty infinite loop) by inserting llvm.sideeffect"),
     deduplicate_diagnostics: Option<bool> = (None, parse_opt_bool, [UNTRACKED],
         "deduplicate identical diagnostics"),
+    lazy_normalization: bool = (false, parse_bool, [UNTRACKED],
+        "lazily evaluate constants (experimental)"),
```

Comment:
`-Z` flags are less convenient to experiment with, e.g. on the playground (to find bugs and whatnot), so why not use a feature gate?

Reply:
It might be useful to have both. A `-Z` flag will allow turning on lazy normalization for an entire crate graph, via `RUSTFLAGS`

Reply:
My thinking was that this doesn't just affect code that uses the const generic feature, so having a `-Z` flag would be useful to test non-feature gated crates. Though enabling it also with a feature gate would also be good, as I had to partially already do that to get the impls inside the standard library that use const generics to work.

Reply:
`tidy` is complaining that there is no gate test for this feature, how would I write a gate test for this? Detecting when this feature is needed is very hard, as evident by the previously poor diagnostics/ICEs.

Also is there a way to enable a feature via `RUSTFLAGS`? I don't want to check for both whether the feature is enabled and the debugging option, every time I check if lazy normalization is enabled.